sql/size_utils.sql

```diff
@@ -682,3 +635,126 @@ BEGIN
 END;
 $BODY$;
 
+-------------Get index size for hypertables -------
+-- hypertable_index_name      - index on hyper table
+CREATE OR REPLACE FUNCTION _timescaledb_internal.indexes_local_size(
+    index_name_in              NAME
+)
+RETURNS TABLE ( hypertable_id INTEGER,
+                total_bytes BIGINT ) 
+               LANGUAGE PLPGSQL STABLE STRICT
+               AS
+$BODY$
+DECLARE
+  sumval BIGINT;
+BEGIN
+        RETURN QUERY
+        SELECT ci.hypertable_id, sum(pg_relation_size(c.oid))::bigint
+--        INTO sumval
+        FROM                                      
+        pg_class c,
+        pg_namespace n,
+        _timescaledb_catalog.hypertable h,
+        _timescaledb_catalog.chunk ch,
+        _timescaledb_catalog.chunk_index ci
+        WHERE ch.schema_name = n.nspname
+            AND c.relnamespace = n.oid
+            AND c.relname = ci.index_name
+            AND ch.id = ci.chunk_id
+            AND h.id = ci.hypertable_id
+            AND ci.hypertable_index_name = index_name_in
+        GROUP BY ci.hypertable_id; 
+END;
+$BODY$;
+
+CREATE OR REPLACE FUNCTION _timescaledb_internal.data_node_index_size (node_name name, index_name_in name)
+RETURNS TABLE ( hypertable_id INTEGER, total_bytes BIGINT)
+AS '@MODULE_PATHNAME@' , 'ts_dist_remote_hypertable_index_info' LANGUAGE C VOLATILE STRICT;
+
+CREATE OR REPLACE FUNCTION _timescaledb_internal.indexes_remote_size(
+    schema_name_in             NAME,
+    table_name_in              NAME,
+    index_name_in              NAME
+)
+RETURNS TABLE (
+    total_bytes bigint,
+    node_name   NAME)
+               LANGUAGE PLPGSQL STABLE STRICT
+               AS
+$BODY$
+BEGIN
+    RETURN QUERY
+    SELECT
+        sum(entry.total_bytes)::bigint AS total_bytes,
+        srv.node_name
+    FROM (
+        SELECT
+            s.node_name,
+            _timescaledb_internal.ping_data_node (s.node_name) AS node_up
+        FROM
+            _timescaledb_catalog.hypertable AS ht,
+            _timescaledb_catalog.hypertable_data_node AS s
+        WHERE
+            ht.schema_name = schema_name_in
+            AND ht.table_name = table_name_in
+            AND s.hypertable_id = ht.id
+         ) AS srv
+    LEFT OUTER JOIN LATERAL _timescaledb_internal.data_node_index_size(
```

Comment:
Is there any sense to have outer joint, since it is not meaningful to ask for size on data node with NULL name? Can it be just?:
 ```suggestion
    JOIN LATERAL _timescaledb_internal.data_node_index_size(
```

Reply:
does not make a difference anymore since the function signature changed and we are aggregating the result.